### PR TITLE
Implemented: functionality to product store card on facility details page (#10)

### LINKED ISSUE #10: Make the product store card dynamic and functional on the facility details page
## What are the acceptance criteria?

- [ ] Display all product stores inside the modal
- [ ] Remove product store association with facility
- [ ] Make a product store as primary for facility
- [ ] Associate a product store with facility

## Can you complete this feature request by yourself?

- [x] YES
- [ ] NO

## Additional information

https://www.figma.com/file/bVPRRw282CqGKMdbz7dciH/HC-Ionic-design-system?type=design&node-id=26705-86775&mode=design&t=W2MtoUb8vPt9MgQj-0

### PR DESCRIPTION
**Pull request by amansinghbais**
### Related Issues

Closes #10

### Short Description and Why It's Useful

Made product store card dynamic and functional on the facility details page.
-  Displayed all product stores inside the modal
-  Displayed associated product stores in the product store card at facility details page.
-  Api and functionality to remove product store association with facility.
-  Api and functionality to associate a product store with facility.

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/facilities#contribution-guideline)

### CODE REVIEW

File: src/store/modules/facility/actions.ts [hunk 1]
```diff
@@ -141,6 +141,33 @@ const actions: ActionTree<FacilityState, RootState> = {
...
+      entityName: 'ProductStoreFacility',
+      filterByDate: 'Y',
+      fieldList: ['fromDate', 'productStoreId']
+    }
+
+    try {
+      const resp = await FacilityService.getFacilityProductStores(payload)
+
+      if (!hasError(resp) || resp.data.error === 'No record found') {
+        productStores = resp.data.docs ? resp.data.docs : []
```
Comment (ymaheshwari1) [inline src/store/modules/facility/actions.ts:162]: No need to use ternary, as we will already check whether the resp has some data or not, using count param

File: src/store/modules/facility/actions.ts [hunk 2]
```diff
@@ -141,6 +141,33 @@ const actions: ActionTree<FacilityState, RootState> = {
...
+      } else {
+        throw resp.data
+      }
+    } catch (error) {
+      console.error('Failed to fetch user associated product stores.', error)
```
Comment (ymaheshwari1) [inline src/store/modules/facility/actions.ts:167]: Remove unwanted console statements

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/FacilityDetails.vue [hunk 3]
```diff
@@ -140,27 +140,17 @@
               <ion-card-title>
                 {{ translate("Product Stores") }}
               </ion-card-title>
-              <ion-button @click="addProductStore" fill="clear">
+              <ion-button @click="selectProductStore()" fill="clear">
```
Comment (ymaheshwari1) [inline src/views/FacilityDetails.vue:143]:
```suggestion
              <ion-button @click="selectProductStores()" fill="clear">
```

File: src/views/FacilityDetails.vue [hunk 4]
```diff
@@ -522,12 +527,53 @@ export default defineComponent({
...
+            .map(async (payload: any) => await FacilityService.createProductStoreFacility({
+              productStoreId: payload.productStoreId,
+              facilityId: this.facilityId,
+              fromDate: DateTime.now().toMillis(),
+            }))
+          )
+
+          const hasFailedResponse = [...updateResponses, ...createResponses].some((response: any) => response.status === 'rejected')
+          if(hasFailedResponse) {
+            showToast(translate('Failed to update some facility stores'))
```
Comment (ymaheshwari1) [inline src/views/FacilityDetails.vue:560]:
```suggestion
            showToast(translate('Failed to update product stores'))
```

File: src/views/FacilityDetails.vue [hunk 5]
```diff
@@ -522,12 +527,53 @@ export default defineComponent({
...
+          } else {
+            productStoresToRemove.map((store: any) => {
+              if(store.productStoreId === this.primaryMember.facilityGroupId) {
+                this.removeProductFromPrimaryMember()
+              }
+            })
+            showToast(translate('Facility stores updated successfully.'))
```
Comment (ymaheshwari1) [inline src/views/FacilityDetails.vue:567]:
```suggestion
            showToast(translate('Product stores updated successfully.'))
```

File: src/components/ProductStorePopover.vue [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+      return facilityGroupId
+    },
+    async removeProductFromPrimaryMember() {
+      let resp;
+      try {
+        resp = await FacilityService.updateFacilityToGroup({
+          "facilityId": this.facilityId,
+          "facilityGroupId": this.primaryMember.facilityGroupId,
+          "fromDate": this.primaryMember.fromDate,
+          "thruDate": DateTime.now().toMillis()
```
Comment (ymaheshwari1) [inline src/components/ProductStorePopover.vue:172]: Handle case when having error in the resp.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/ProductStorePopover.vue [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+    </ion-list>
+  </ion-content>
+</template>
+
+<script lang="ts">
+import {
+  IonContent,
+  IonList,
+  IonListHeader,
+  IonIcon,
```
Comment (ravilodhi) [inline src/components/ProductStorePopover.vue:22]: Arrange imports in alphabetical order.

**Pull request merged**